security/WFCORE-5466_Remove_Legacy_Vault_Integration.adoc

```diff
@@ -0,0 +1,311 @@
+
+* An early PR to WildFly to compensate for forthcoming changes.
+* Submit main PR to WildFly Core removing the vault integration.
+* A third PR to WildFly completing the removal including documentation updates.
+
+During this task follow up activities will be identified, these should not block the progress
+of the initial PRs being merged.
+
+Follow up tasks may include:
+
+* Re-purposing test cases or developing new test cases where Elytron coverage is required.
+* Additional enhancements for feature parity in relation to removed features.
+
+== Test Plan
+
+The following table identifies the tests in WildFly Core and WildFly affected by the removal.
+
+.Test Case Updates
+|===
+|Project |Test Case |Action
+
+|WildFly Core
+|org.jboss.as.server.services.security.RuntimeVaultReaderUnitTestCase
+|Removed
+
+|WildFly Core
+|org.jboss.as.server.RuntimeExpressionResolverUnitTestCase
+|Removed
+
+|WildFly Core
+|org.jboss.as.server.deployment.DeploymentAddHandlerTestCase
+|Tweaked
+
+|WildFly Core
+|org.jboss.as.server.test.InterfaceManagementUnitTestCase
+|Tweaked
+
+|WildFly Core
+|org.jboss.as.server.parsing.SystemPropertiesParsingTest
+|Reduced
+
+|WildFly Core
+|org.jboss.as.domain.controller.operations.SyncModelServerStateTestCase
+|Tweaked
+
+|WildFly Core
+|org.jboss.as.core.model.test.vault.HostVaultTestCase
+|Removed
+
+|WildFly Core
+|org.jboss.as.core.model.test.vault.StandaloneVaultTestCase
+|Removed
+
+|WildFly Core
+|org.jboss.as.core.model.test.host.HostModelTestCase#testDefaultHostXmlWithExpressions
+|Tweaked
+
+|WildFly Core
+|org.wildfly.core.test.standalone.mgmt.api.core.MissingVaultExpressionTestCase
+|Removed
+
+|WildFly Core
+|org.jboss.as.test.manualmode.vault.VaultPasswordsInCLITestCase
+|Removed
+
+|WildFly Core
+|org.jboss.as.test.manualmode.vault.CustomVaultInModuleTestCase
+|Removed
+
+|WildFly Core
+|org.jboss.as.test.manualmode.management.cli.CustomVaultInCLITestCase
+|Removed
+
+|WildFly Core
+|org.jboss.as.test.integration.mgmt.access.VaultExpressionSensitivityTestCase
+|Removed
+
+|WildFly Core
+|org.jboss.as.test.integration.mgmt.access.ValidateAddressOrOperationTestCase
+|Reduced
+
+|WildFly Core
+|org.jboss.as.test.integration.domain.SlaveHostControllerAuthenticationTestCase
+|Reduced
+
+|WildFly Core
+|org.jboss.as.test.integration.domain.rbac.AccessConstraintUtilizationTestCase
+|Tweaked
+
+|WildFly
+|org.jboss.as.test.integration.security.passwordmasking.PasswordMaskingInContainerTestCase
+|Removed
+
+|WildFly
+|org.jboss.as.test.integration.security.passwordmasking.PasswordMaskingTestCase
+|Removed
+
+|WildFly
+|org.jboss.as.test.integration.security.loginmodules.LdapExtLoginModuleTestCase
+|Reduced
+
+|WildFly
+|org.jboss.as.test.integration.messaging.jms.definitions.JMSResourceDefinitionsTestCase
+|Reduced
+
+|WildFly
+|org.jboss.as.test.integration.messaging.jms.context.VaultedInjectedJMSContextTestCase
+|Reduced
+
+|WildFly
+|org.jboss.as.test.integration.security.vault.ExternalPasswordCommandsTestCase
```

Comment:
I cannot find an alternative test case for credential stores. We should add a followup task for this.

Reply:
The test case covers also usage of Java class as an external source of master password. There seems to be no alternative for credential store - do we want to provide such feature?

https://access.redhat.com/documentation/en-us/red_hat_jboss_enterprise_application_platform/7.4/html-single/how_to_configure_server_security/index#obtain_keystore_password_from_external_source `{CLASS[@jboss_module_spec]}classname[:ctorargs]`